Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -64,47 +68,56 @@ public class CustomerIOImplementation: CustomerIOInstance {
+            // TODO: add to background queue delete device token from currently registered profile.
+        }
+
+        let jsonBodyString = jsonAdapter.toJsonString(body, encoder: jsonEncoder)
+
+        let queueTaskData = IdentifyProfileQueueTaskData(identifier: identifier,
+                                                         attributesJsonString: jsonBodyString)
+        let queueStatus = backgroundQueue.addTask(type: .identifyProfile, data: queueTaskData)
+
+        // don't modify the state of the SDK until we confirm we added a background queue task successfully.
+        if queueStatus.success {
+            profileStore.identifier = identifier
+        }
+
+        // TODO: after background queue fully implemented into the whole SDK (not just Tracking module) I see
+        // this no longer being needed. This is currently the way for other SDKs (modules) in the project at runtime
+        // to get informed about something happening but we plan on instead changing to runtime "hooks" between
+        // the SDKs.
+        // self.eventBus.post(.identifiedCustomer)
     }
 
     public func clearIdentify() {
-        identifyRepository.removeCustomer()
+        profileStore.identifier = nil
     }
 
     public func track<RequestBody: Encodable>(
         name: String,
         data: RequestBody,
-        jsonEncoder: JSONEncoder? = nil,
-        onComplete: @escaping (Result<Void, CustomerIOError>) -> Void
+        jsonEncoder: JSONEncoder? = nil
     ) {
-        // XXX: once we have a bg queue, if this gets deferred to later we should set a timestamp value
-        identifyRepository
-            .trackEvent(name: name, data: data, timestamp: nil, jsonEncoder: jsonEncoder) { [weak self] result in
-                DispatchQueue.main.async { [weak self] in
-                    guard self != nil else { return }
-
-                    switch result {
-                    case .success:
-                        return onComplete(Result.success(()))
-                    case .failure(let error):
-                        return onComplete(Result.failure(error))
-                    }
-                }
-            }
+        guard let currentlyIdentifiedProfileIdentifier = profileStore.identifier else {
+            // XXX: when we have anonymous profiles in SDK,
+            // we can decide to not ignore events when a profile is not logged yet.
+            return
+        }
+
+        let requestBody = TrackRequestBody(name: name, data: data, timestamp: Date())
+        guard let jsonBodyString = jsonAdapter.toJsonString(requestBody, encoder: jsonEncoder) else {
+            // XXX: log error for customer to debug their request body
+            return
+        }
+
+        let queueData = TrackEventQueueTaskData(identifier: currentlyIdentifiedProfileIdentifier,
+                                                attributesJsonString: jsonBodyString)
+
+        // ignore if adding task was successful or not.
```

Comment:
I agree that errors should not be ignored in the SDK. However, I believe that is a much bigger task then this background queue. I think that we need to come up with a strategy for error handling in the SDK in general, not just when an error happens in the queue. 

When adding tasks to the queue, the error that can happen is saving to the file system had an error. That could be because the device is out of storage space of there is a bug in the SDK code causing the file system to not be able to save successfully. 

With that use case, I do not believe something like an `onComplete` handler should be used but more of a SDK wide error handling setup.

Reply:
I'd suggest tagging with a `TODO` to make easier / clearer that we're planning to better support this then :+1: